ci/kokoro/docker/build.sh

```diff
@@ -514,6 +514,13 @@ if [[ -n "${KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH:-}" ]]; then
   docker_flags+=("--env" "KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH=${KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH:-}")
 fi
 
+# Optionally allow the docker container to use the host's networking in order
+# to take advantage of potentially non-trivial network configurations (e.g.,
+# DNS cache) that might be a PITA to duplicate within the container itself.
```

Comment:
If this doesn't deserve to be the default, we should mention why (e.g., the [current] use of a fixed emulator port).

Aside: There is also the possibility of making the choice automatically if we can predict when the container will use a rate-limited DNS server.  But in the end it might be nicer if we can indeed figure out how to run a caching domain server in the container, or how to forward just the DNS traffic to the host caching server.

Aside2: We should probably also keep in mind that customers not using a DNS cache may see the same issue.  And even if they do have a cache, they may see rate limiting when actually connecting to the target service (as we saw when running many tests quickly).

Reply:
> If this doesn't deserve to be the default, we should mention why (e.g., the [current] use of a fixed emulator port).
> 
> Aside: There is also the possibility of making the choice automatically if we can predict when the container will use a rate-limited DNS server.

Not sure how to do that generically.

> But in the end it might be nicer if we can indeed figure out how to run a caching domain server in the container,

That'd work.

> or how to forward just the DNS traffic to the host caching server.

Done. That was my first "hacky" solution.

```shell
sudo socat UDP-LISTEN:53,bind=192.168.9.1,reuseaddr,fork UDP:127.0.0.1:53
```

Which binds my `docker0`'s IP (192.168.9.1) and forwards everything to localhost:53. Well, actually, this only does UDP; we should forward TCP also to handle DNS more robustly. But any forwarding solution like this needs to be set up before we start the docker container, which can get a little complicated (e.g., the above solution requires that the local machine has `socat` installed).

> 
> Aside2: We should probably also keep in mind that customers not using a DNS cache may see the same issue. And even if they do have a cache, they may see rate limiting when actually connecting to the target service (as we saw when running many tests quickly).

These sound mostly like application-level issues.